libc/string.c

```diff
@@ -0,0 +1,184 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libc-tests
+ *
+ * Testing string.h functions
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <string.h>
+
+#include <unity_fixture.h>
+
+/* fixtures for strlcpy */
+char source[5];
+char dest[5];
+char tmplt[5];
+
+/* fixtures for strlcat */
+char source1[4];
+char source2[6];
+char buffer[12];
+
+/* common fixtures */
+int i;
+
+
+TEST_GROUP(string_strlcpy);
+
+
+TEST_SETUP(string_strlcpy)
+{
+	memcpy(source, "abcd", 5);
+	memcpy(dest, "xxxx", 5);
+	i = -1;
+}
+
+
+TEST_TEAR_DOWN(string_strlcpy)
+{
+}
+
+
+TEST(string_strlcpy, strlcpy_fullcopy)
+{
+	/* Test full copy */
+	i = strlcpy(dest, source, 5);
+	TEST_ASSERT_EQUAL_INT(4, i);
+	TEST_ASSERT_EQUAL_STRING(source, dest);
+}
+
+
+TEST(string_strlcpy, strlcpy_shorter)
+{
+	/* Test shorter than source copy */
+	i = strlcpy(dest, source, 3);
+	TEST_ASSERT_EQUAL_STRING("ab", dest);
+	TEST_ASSERT_EQUAL_INT(4, i);
+}
+
+
+TEST(string_strlcpy, strlcpy_longer)
+{
+	/* Test longer than source copy */
+	source[3] = '\0'; /* source is now "abc" null terminated; */
+	i = strlcpy(dest, source, 5);
+	TEST_ASSERT_EQUAL_STRING("abc", dest);
+	TEST_ASSERT_EQUAL_INT(3, i);
+	source[3] = 'd';
+}
+
+
+TEST(string_strlcpy, strlcpy_onelength)
+{
+	/* Test 1 length copy */
+	i = strlcpy(dest, source, 1);
+	TEST_ASSERT_EQUAL_STRING("", dest);
+	TEST_ASSERT_EQUAL_INT(4, i);
+}
+
+
+TEST(string_strlcpy, strlcpy_zerolength)
+{
+	/* Test 1 length copy */
+	i = strlcpy(dest, source, 0);
+	TEST_ASSERT_EQUAL_STRING("xxxx", dest);
+	TEST_ASSERT_EQUAL_INT(4, i);
+}
+
+
+TEST_GROUP_RUNNER(string_strlcpy)
+{
+	RUN_TEST_CASE(string_strlcpy, strlcpy_fullcopy);
+	RUN_TEST_CASE(string_strlcpy, strlcpy_shorter);
+	RUN_TEST_CASE(string_strlcpy, strlcpy_longer);
+	RUN_TEST_CASE(string_strlcpy, strlcpy_onelength);
+	RUN_TEST_CASE(string_strlcpy, strlcpy_zerolength);
+}
+
+
+TEST_GROUP(string_strlcat);
+
+
+TEST_SETUP(string_strlcat)
+{
+	memcpy(source1, "abc", 4);
+	memcpy(source2, "defgh", 6);
+	memcpy(buffer, "klmnopqrstu", 12);
+	i = -1;
```

Comment:
I don't think we need this - it's always assigned by each test

Reply:
Yes, you are right. Removed `i = -1`. Thanks!